tests/integration/components/exclaim-ui-test.js

```diff
@@ -116,6 +117,32 @@ test('it writes bound data back to the env', function(assert) {
   assert.equal(get(this, 'env.envValue'), 'after');
 });
 
+test('it should call the onChange action when the env changes', function(assert) {
+  set(this, 'onChange', sinon.spy());
+
+  getOwner(this).register('component:simple-component', Component.extend({
+    layout: hbs`<input value={{config.value}} oninput={{action (mut config.value) value='target.value'}}>`
+  }));
+
+  set(this, 'ui', {
+    $component: 'simple-component',
+    value: {
+      $bind: 'envValue'
+    }
+  });
+
+  set(this, 'env', {
+    envValue: 'before'
+  });
+
+  this.renderUI();
+
+  run(() => fillIn('input', 'after'))
+
+  assert.ok(get(this, 'onChange').calledOnce);
+  assert.ok(get(this, 'onChange').calledWith({ key: 'envValue', value: 'after' }));
```

Comment:
I wonder if we actually need to send the value up? I could imagine the signature just being `calledWith('envValue')`

Reply:
I suppose not, you could just get the value off of the environment since you'll have the key